src/dev/precommit_hook/casing_check_config.js

```diff
@@ -53,7 +53,7 @@ export const IGNORE_FILE_GLOBS = [
  * @type {Array}
  */
 export const KEBAB_CASE_DIRECTORY_GLOBS = [
-  'packages/*',
+  'packages/!(__mocks__)',
```

Comment:
According to Jest, if the module you are mocking is a Node module, the mock should be placed in the `__mocks__` directory. At the same time there is a rule in kibana in which files under packages should use kebab case.
Since mocking the `@kbn/i18n` module is a workaround, after merging the PR #20513 I will revert this change.